pkg/virt-operator/kubevirt.go

```diff
@@ -659,17 +661,30 @@ func (c *KubeVirtController) execute(key string) error {
 	// set timestamps on conditions if they changed
 	operatorutil.SetConditionTimestamps(kv, kvCopy)
 
-	// If we detect a change on KubeVirt we update it
+	// If we detect a change on KubeVirt status we update it
 	if !reflect.DeepEqual(kv.Status, kvCopy.Status) {
-		if err := c.statusUpdater.UpdateStatus(kvCopy); err != nil {
-			logger.Reason(err).Errorf("Could not update the KubeVirt resource status.")
+		statusJson, err := json.Marshal(kvCopy.Status)
+		if err != nil {
+			return err
+		}
+		patch := fmt.Sprintf(`[{"op": "replace", "path": "/status", "value": %s}]`, string(statusJson))
```

Comment:
Since we own the object, I think that here it is fine to use `UpdateStatus`. But if we hot-loop and we always patch or update the status, then we will always fail updating the finalizers below because the resource version changed. This is what would really be interesting to understand.

Reply:
yeah agreed. The patch is really only necessary for the finalizer.

Definitely interested to understand why it appears we update the status every reconcile loop.